tests/python/pants_test/engine/test_scheduler.py

```diff
@@ -270,29 +270,19 @@ def test_get_type_match_failure(self):
     def test_unhashable_failure(self):
         """Test that unhashable Get(...) params result in a structured error."""
 
-        def assert_has_cffi_extern_traceback_header(exc_str):
-            self.assertTrue(
-                exc_str.startswith(
-                    dedent(
-                        """\
-                        1 Exception raised in CFFI extern methods:
-                        Traceback (most recent call last):
-                        """
-                    )
-                ),
-                f"exc_str was: {exc_str}",
-            )
-
-        def assert_has_end_of_cffi_extern_error_traceback(exc_str):
-            self.assertIn(
+        def assert_has_cffi_extern_traceback_header(exception: str) -> None:
+            assert exception.startswith(
                 dedent(
                     """\
+                    1 Exception raised in CFFI extern methods:
                     Traceback (most recent call last):
-                      File LOCATION-INFO, in identify
-                        hash_ = hash(obj)
```

Comment:
It's probably good to keep this bit to ensure that `hash` is the issue. Additionally, you may want to assert that `TypeError: unhashable type: 'list'` is in the message.